sdk/ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_17/PrefetchContractKeysIT.scala

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.suites.v1_17
+
+import com.daml.ledger.api.testtool.infrastructure.Allocation.{
+  Participant,
+  Participants,
+  SingleParty,
+  allocate,
+}
+import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
+import com.daml.ledger.api.testtool.suites.v1_8.CompanionImplicits
+import com.daml.ledger.api.v1.commands
+import com.daml.ledger.javaapi.data.PrefetchContractKey
+import com.daml.ledger.test.java.model.da.types.Tuple2
+import com.daml.ledger.test.java.model.test.{Dummy, TextKey, TextKeyOperations, WithKey}
+
+import scala.jdk.CollectionConverters._
+import scala.jdk.OptionConverters._
+
+object PrefetchContractKeysIT {
+  implicit final class JavaBindingSupportExtension(val prefetch: PrefetchContractKey)
+      extends AnyVal {
+    def toProtoInner: commands.PrefetchContractKey = {
+      commands.PrefetchContractKey.fromJavaProto(prefetch.toProto)
+    }
+  }
+}
+
+class PrefetchContractKeysIT extends LedgerTestSuite {
+  import CompanionImplicits._
+  import PrefetchContractKeysIT._
+
+  test(
+    "CSprefetchContractKeysBasic",
+    "Explicit contract key prefetches are accepted",
+    allocate(SingleParty),
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    val prefetch = WithKey.byKey(party).toPrefetchKey().toProtoInner
+    val request = ledger
+      .submitAndWaitRequest(party, new WithKey(party).create.commands)
```

Comment:
I think we could create an invalid key if we went a la nature then we would be able to see the transaction getting rejected WDYT?

Reply:
Added it in 93ff12b7a6249d2ec9a01696ac8994c436dfba2b